processor/samplingprocessor/processor.go

```diff
@@ -18,79 +18,201 @@ import (
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottllog"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlspan"
 	"go.opentelemetry.io/collector/pdata/plog"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.opentelemetry.io/collector/pdata/ptrace"
 	"go.uber.org/zap"
 )
 
-type samplingProcessor struct {
+type logsSamplingProcessor struct {
 	logger          *zap.Logger
 	dropCutOffRatio float64
+	condition       *expr.OTTLCondition[ottllog.TransformContext]
 }
 
-func newSamplingProcessor(logger *zap.Logger, cfg *Config) *samplingProcessor {
-	return &samplingProcessor{
+type metricsSamplingProcessor struct {
+	logger          *zap.Logger
+	dropCutOffRatio float64
+	condition       *expr.OTTLCondition[ottldatapoint.TransformContext]
+}
+
+type tracesSamplingProcessor struct {
+	logger          *zap.Logger
+	dropCutOffRatio float64
+	condition       *expr.OTTLCondition[ottlspan.TransformContext]
+}
+
+func newLogsSamplingProcessor(logger *zap.Logger, cfg *Config, condition *expr.OTTLCondition[ottllog.TransformContext]) *logsSamplingProcessor {
+	return &logsSamplingProcessor{
+		logger:          logger,
+		dropCutOffRatio: cfg.DropRatio,
+		condition:       condition,
+	}
+}
+
+func newMetricsSamplingProcessor(logger *zap.Logger, cfg *Config, condition *expr.OTTLCondition[ottldatapoint.TransformContext]) *metricsSamplingProcessor {
+	return &metricsSamplingProcessor{
+		logger:          logger,
+		dropCutOffRatio: cfg.DropRatio,
+		condition:       condition,
+	}
+}
+
+func newTracesSamplingProcessor(logger *zap.Logger, cfg *Config, condition *expr.OTTLCondition[ottlspan.TransformContext]) *tracesSamplingProcessor {
+	return &tracesSamplingProcessor{
 		logger:          logger,
 		dropCutOffRatio: cfg.DropRatio,
+		condition:       condition,
 	}
 }
 
-func (sp *samplingProcessor) sampleFunc() bool {
+func (sp *logsSamplingProcessor) sampleFunc() bool {
 	//#nosec G404 -- randomly generated number is not used for security purposes. It's ok if it's weak
 	return rand.Float64() <= sp.dropCutOffRatio
 }
 
-func (sp *samplingProcessor) processTraces(_ context.Context, td ptrace.Traces) (ptrace.Traces, error) {
+func (sp *metricsSamplingProcessor) sampleFunc() bool {
+	//#nosec G404 -- randomly generated number is not used for security purposes. It's ok if it's weak
+	return rand.Float64() <= sp.dropCutOffRatio
+}
+
+func (sp *tracesSamplingProcessor) sampleFunc() bool {
+	//#nosec G404 -- randomly generated number is not used for security purposes. It's ok if it's weak
+	return rand.Float64() <= sp.dropCutOffRatio
+}
+
+func (sp *tracesSamplingProcessor) processTraces(ctx context.Context, td ptrace.Traces) (ptrace.Traces, error) {
 	switch {
-	case sp.dropCutOffRatio == 1.0: // Drop everything
-		return ptrace.NewTraces(), nil
 	case sp.dropCutOffRatio == 0.0: // Drop nothing
 		return td, nil
-	default: // Drop based on ratio
+	default: // Drop based on ratio and condition
```

Comment:
For these process funcs, can we refactor the switch statement to an if conditional for simplicity
```
if sp.dropCutOffRatio == 0.0 {
    return td, nil
}
....
```